tokio-util/src/io/sync_bridge.rs

```diff
@@ -6,6 +6,81 @@ use tokio::io::{
 
 /// Use a [`tokio::io::AsyncRead`] synchronously as a [`std::io::Read`] or
 /// a [`tokio::io::AsyncWrite`] as a [`std::io::Write`].
+///
+/// # Alternatives
+///
+/// In many cases, there are better alternatives to using `SyncIoBridge`, especially if you
+/// want to avoid blocking the async runtime. Consider the following scenarios:
+///
+/// ## Example 1: Hashing Data
+///
+/// When hashing data, using `SyncIoBridge` can lead to suboptimal performance and might not fully leverage the async capabilities of the system.
+/// Instead, consider reading the data into memory and then hashing it, or processing the data in chunks.
+///
+/// ```rust, no_run
+/// let mut data = Vec::new();
+/// reader.read_to_end(&mut data).await?;
```

Comment:
For async/await to work in the docs you actually also need a runtime since the doc test will be run by a synchronous main function by default. 

See this from tokio as an example:
```
    /// ```rust,no_run
    /// use std::error::Error;
    ///
    /// #[tokio::main]
    /// async fn main() -> Result<(), Box<dyn Error>> {
    ///     let tokio_listener = tokio::net::TcpListener::bind("127.0.0.1:0").await?;
    ///     let std_listener = tokio_listener.into_std()?;
    ///     std_listener.set_nonblocking(false)?;
    ///     Ok(())
    /// }
    /// ```
```